packages/alfa-rectangle/src/rectangle.ts

```diff
@@ -86,28 +144,76 @@ export class Rectangle
     );
   }
 
-  equals(value: this): boolean;
-  equals(value: unknown): value is this;
-  equals(value: unknown): boolean {
+  public union(other: Rectangle): Rectangle {
+    if (this.contains(other)) {
+      return this;
+    }
+
+    if (other.contains(this)) {
+      return other;
+    }
+
+    const minLeft = min(this.left, other.left);
+    const minTop = min(this.top, other.top);
+    const maxRight = max(this.right, other.right);
+    const maxBottom = max(this.bottom, other.bottom);
+
+    return Rectangle.of(
+      minLeft,
+      minTop,
+      maxRight - minLeft,
+      maxBottom - minTop
+    );
+  }
+
+  public intersection(other: Rectangle): Rectangle {
+    if (!this.intersects(other)) {
+      return Rectangle.empty();
+    }
+
+    if (this.contains(other)) {
+      return other;
+    }
+
+    if (other.contains(this)) {
+      return this;
+    }
```

Comment:
**Question:** are these early escape really needed?

Reply:
I think I had to add the first check due to how the intersection is calculated below which for disjoint rectangles would actually otherwise return a rectangle sitting between the disjoint rectangles. That might on the other hand also be a sign that it's not the best way to calculate it. I'm not sure.